dowhy/gcm/feature.py

```diff
@@ -0,0 +1,251 @@
+                                                       min(max_num_background_samples, feature_samples.shape[0]),
+                                                       replace=False)]
+
+    return feature_relevance_sample(prediction_method,
+                                    feature_samples,
+                                    samples_of_interest,
+                                    feature_attribution_func,
+                                    None,
+                                    True,
+                                    max_batch_size,
+                                    randomize_features_jointly,
+                                    shapley_config)
+
+
+def feature_relevance_sample(prediction_method: Callable[[np.ndarray], np.ndarray],
+                             feature_samples: np.ndarray,
+                             samples_of_interest: np.ndarray,
+                             feature_attribution_func: Callable[[np.ndarray, np.ndarray],
+                                                                Union[np.ndarray, float]],
+                             baseline_values: Optional[np.ndarray] = None,
+                             average_set_function: bool = False,
+                             max_batch_size: int = 100,
+                             randomize_features_jointly: bool = True,
+                             shapley_config: Optional[ShapleyConfig] = None) -> np.ndarray:
+    """ Estimates the feature relevance of the prediction_method for each sample in samples_of_interest. This
+    method uses all samples given in feature_samples as 'background' samples. This is, they should represent samples
+    from the joint distribution of the input features. The feature_attribution_func defines the comparison between the
+    output of the prediction_method when certain features are randomized and the outputs when no features are
+    randomized. The most common function would be the difference between the expectations.
+
+    If the randomized predictions should rather be compared to the original data, this has (and can) be defined via
+    the set function by ignoring the second input parameter (the predicted values using all feauters). Instead, the
+    original data can be used.
+
+    Related paper:
+    Janzing, D., Minorics, L., & Bloebaum, P. (2020).
+    Feature relevance quantification in explainable AI: A causal problem.
+    In International Conference on Artificial Intelligence and Statistics (pp. 2907-2916). PMLR.
+
+    :param prediction_method: A callable that is expected to return a prediction for given samples.
+    :param feature_samples: Samples from the joint distribution. These are used as 'background samples'.
+    :param samples_of_interest: Samples for the feature relevance should be estimated.
+    :param feature_attribution_func: Set function for estimating the attributions of the model outcomes. This function
```

Comment:
It is the set functions used for estimating the Shapley values, i.e. it defines what we are attributing to the inputs. But I see the point that the name ```feature_attribution_func``` is also not really capturing this. Probably the most accurate name would be ```set_function```, but it might be even more confusing. Maybe @kailashbuki has another idea for a name.

Reply:
@amit-sharma What `feature_attribution_func` represents there is what is known as "characteristic function" in [coalition games](http://www.library.fa.ru/files/roth2.pdf), which is defined for every subset of N players we want to attribute some quantity. It is `v` in Patrick's comment [here](https://github.com/py-why/dowhy/pull/450#discussion_r904014288). Some [papers](http://proceedings.mlr.press/v119/sundararajan20b/sundararajan20b.pdf) hence call it "set function". In light of this discussion, I could think of sth like `subset_scoring_func` as a suitable substitute. But I also don't have a strong opinion.

Reply:
Ok, I will change it to ```subset_scoring_func```.

Reply:
thanks, for explaining, @kailashbuki . `subset_scoring_func` is good. Would it make sense to write in the docstring that it is equivalent to the "characteristic function" in coalition games. Perhaps a better place to bring out this connection will be in the proposed "guide to Shapley estimator" docs.

Reply:
I think it makes sense to add this as a brief remark in the docstring(s) as well. I will add it.